### PR TITLE
Add tests for AI class

### PR DESCRIPTION
**Pull request by TheoMcCabe**

### CODE REVIEW

File: tests/test_ai.py [hunk 1]
```diff
@@ -1,9 +1,98 @@
...
+    response_messages = ai.start("system prompt", "user prompt", "step name")
+
+    # assert
+    assert response_messages[-1].content == "response1"
+
+
+def test_next(monkeypatch):
+    """
+    Test function for the AI system.
+    This test sets up a fake LLM and tests that the start method successfully returns a response
```
Comment (TheoMcCabe) [inline tests/test_ai.py:31]: remove repeated comments

File: gpt_engineer/core/ai.py [hunk 2]
```diff
@@ -463,69 +463,63 @@ def num_tokens_from_messages(self, messages: List[Message]) -> int:
         n_tokens += 2  # every reply is primed with <im_start>assistant
         return n_tokens
 
+    def _check_model_acess_and_fallback(self):
```
Comment (ATheorell) [inline gpt_engineer/core/ai.py:466]: It should be access right?
Apart from that, happy to merge

File: gpt_engineer/core/ai.py [hunk 3]
```diff
@@ -463,69 +463,63 @@ def num_tokens_from_messages(self, messages: List[Message]) -> int:
...
+                streaming=True,
+            )
 
-    Parameters
-    ----------
-    model : str
-        The name of the model to create.
-    temperature : float
-        The temperature to use for the model.
+        self._check_model_acess_and_fallback()
```
Comment (ATheorell) [inline gpt_engineer/core/ai.py:515]: here too

File: gpt_engineer/core/ai.py (hunk 2 above)
Comment (TheoMcCabe) [reply gpt_engineer/core/ai.py]: yep done

**Pull request merged**